### PR TITLE
hive: clusterimageset added unit tests

### PR DESCRIPTION
**Pull request by kononovn**

### CODE REVIEW

File: pkg/hive/clusterimageset_test.go [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+		},
+	}
+
+	for _, testCase := range testCases {
+		err := testCase.testClusterImageSet.Delete()
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testCase.testClusterImageSet.Object)
+			assert.Nil(t, err)
```
Comment (trewest) [inline pkg/hive/clusterimageset_test.go:244]: Already done on L240

File: pkg/hive/clusterimageset_test.go [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+			releaseImage:        "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testClusterImageSetBuilder := testCase.testClusterImageSet.WithReleaseImage(testCase.releaseImage)
+		if testCase.releaseImage != "" {
+			assert.Equal(t, testCase.releaseImage, testClusterImageSetBuilder.Definition.Spec.ReleaseImage)
+		} else {
+			assert.Equal(t, defaultClusterImageSetRelease, testClusterImageSetBuilder.Definition.Spec.ReleaseImage)
```
Comment (trewest) [inline pkg/hive/clusterimageset_test.go:309]: Not sure I understand this. Shouldn't you have expectedError and check that it is equal to testClusterImageSetBuilder.errorMsg

Comment (kononovn) [reply pkg/hive/clusterimageset_test.go:309]: Yes right. Added:
assert.Equal(t, testCase.expectedError, testClusterImageSetBuilder.errorMsg)

File: pkg/hive/clusterimageset_test.go [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+		assert.NotNil(t, testImageSet.Definition)
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testCase.name, testImageSet.Definition.Name)
+		}
+	}
+}
+
+func TestPullClusterImageSet(t *testing.T) {
+	generateClusterImageSet := func(name string) *hivev1.ClusterImageSet {
```
Comment (achuzhoy) [inline pkg/hive/clusterimageset_test.go:55]: Can this be moved to after the testCases struct for consistency.

File: pkg/hive/clusterimageset_test.go [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+		},
+		{
+			testClusterImageSet: buildValidClusterImageSetBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:       fmt.Errorf("clusterimagesets.hive.openshift.io \"imageset\" not found"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		assert.Equal(t, testCase.testClusterImageSet.Definition.Spec.ReleaseImage, "imageset-release-1-1")
+		testCase.testClusterImageSet.Definition.ResourceVersion = "999"
```
Comment (achuzhoy) [inline pkg/hive/clusterimageset_test.go:206]: redundant?

**Review by trewest — APPROVED** [review]
/lgtm

**Review by cdvultur — APPROVED** [review]
/lgtm

**Pull request merged**